### PR TITLE
receiverID and messageCounter added to isoObject

### PR DESCRIPTION
**Pull request by Angeleon**
Support for receiverID as implemented in iso22133 branch feature_headerInputOutput.

### CODE REVIEW

**Review by ceciliahernqvist — COMMENTED** [review]
Looks good to me, but haven't worked much with the isoObject, so someone else with more knowledge about it should probably also have a look.

**Review by SebastianSvanland — APPROVED** [review]
Looks good

File: tests/isoObject.cpp [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+{
+	// Get system time
+	struct timespec currentTime;
+	clock_gettime(CLOCK_REALTIME, &currentTime);
+	time->tv_sec = currentTime.tv_sec;
+	time->tv_usec = currentTime.tv_nsec / 1000;
+	return time;
+}
+
+class TCPConnection
```
Comment (oscjoha) [inline tests/isoObject.cpp:25]: As I understand this is more of a ATOS emulator? I find it a bit confusing to have a UDP socket inside a class called TCPconnection

File: isoObject.i [hunk 2]
```diff
@@ -37,55 +45,17 @@
...
-} GeographicPositionType;
-
-
-
-typedef struct {
-	double longitudinal_m_s2;
-	double lateral_m_s2;
-	bool isLongitudinalValid;
-	bool isLateralValid;
-} AccelerationType;
```
Comment (oscjoha) [inline isoObject.i:91]: I think you need these to be able to initialize a corresponding type in the target language

File: tests/isoObject.cpp (hunk 1 above)
Comment (seifbourogaa) [reply tests/isoObject.cpp:25]: ^ I second this.

**Review by seifbourogaa — COMMENTED** [review]
Code largely looks good to me, not sure about the SWIG functionality.

File: src/iso22133object.cpp [hunk 3]
```diff
@@ -15,27 +15,50 @@ TestObject::TestObject(const std::string& listenIP)
 	  trajDecoder(),
 	  ctrlChannel(listenIP, ISO_22133_DEFAULT_OBJECT_TCP_PORT),
 	  processChannel(listenIP, ISO_22133_OBJECT_UDP_PORT),
+	  transmitterID(TRANSMITTER_ID_UNAVAILABLE_VALUE),
+	  serverID(0),
+	  expectedMessageCounter(0),
+	  sentMessageCounter(0),
+	  socketsReceivedFromController(true),
```
[the remark this answers is not in the document]
Comment (Angeleon) [reply src/iso22133object.cpp]: tes

[the remark this answers is not in the document]
Comment (Angeleon) [reply src/iso22133object.cpp]: yes

File: CMakeLists.txt [hunk 4]
```diff
@@ -1,5 +1,18 @@
...
+execute_process(
+	COMMAND ping www.github.com -c 1 -q
+	ERROR_QUIET
+	RESULT_VARIABLE NO_CONNECTION
+)
+FetchContent_Declare(googletest
+	GIT_REPOSITORY https://github.com/google/googletest.git
+	GIT_TAG main
+)
+FetchContent_MakeAvailable(googletest)
```
[the remark this answers is not in the document]
Comment (Angeleon) [reply CMakeLists.txt:14]: will do

File: tests/isoObject.cpp (hunk 1 above)
Comment (Angeleon) [reply tests/isoObject.cpp:25]: renamed

File: src/iso22133object.cpp [hunk 5]
```diff
@@ -403,4 +472,14 @@ void TestObject::setNetworkDelay(std::chrono::milliseconds delay) {
 	estimatedNetworkDelay = delay;
 }
 
+void TestObject::checkAndUpdateMessageCounter(const char receivedMessageCounter) {
```
[the remark this answers is not in the document]
Comment (Angeleon) [reply src/iso22133object.cpp:479]: removed

File: inc/iso22133object.hpp [hunk 6]
```diff
@@ -80,8 +80,22 @@ class TestObject {
...
 
-	int handleUDPMessage(char *buffer, int bufferLen, int udpSocket, char* addr, const uint32_t port);
-	int handleTCPMessage(char *buffer, int bufferLen);
+
+	/** SWIG Wrappers **/
+	bool isAwaitingFirstHeab() { return awaitingFirstHeab;}
+	//! Set the endpoint for the process channel
+	void setProcessChannelEndpoint(int udpSocket, char *addr, const uint32_t port);
+	//! Wrapper for handling function that converts to char vector
+	int handleMessage(char *buffer, int bufferLen);
```
Comment (SebastianSvanland) [inline inc/iso22133object.hpp:89]: You have two functions declared with the same name, "handleMessage". Here and on row 181.

**Review by SebastianSvanland — COMMENTED** [review]
Just a single comment, otherwise I think it looks good

File: inc/iso22133object.hpp (hunk 6 above)
Comment (alfaro01) [reply inc/iso22133object.hpp:89]: I'd assume it's overridden?

Comment (alfaro01) [reply inc/iso22133object.hpp:89]: Yeah it's defined correctly in iso22133object.cpp

File: src/iso22133object.cpp [hunk 7]
```diff
@@ -264,6 +306,32 @@ void TestObject::onHeabTimeout() {
...
+int TestObject::handleTCPMessage(const char *buffer, size_t bufLen) {
+	std::vector<char> dataBuffer;
+	for (size_t i = 0; i < bufLen; i++) {
+		dataBuffer.push_back(buffer[i]);
+		printf("%02x ", buffer[i]);
+	}
+	return handleMessage(dataBuffer);
+}
+
+int TestObject::handleUDPMessage(const char *buffer, size_t bufLen, int udpSocket, const std::string &addr, const uint32_t port) {
```
[the remark this answers is not in the document]
Comment (Angeleon) [reply src/iso22133object.cpp]: removed

File: src/iso22133object.cpp [hunk 8]
```diff
@@ -264,6 +306,32 @@ void TestObject::onHeabTimeout() {
 	this->state->handleEvent(*this, Events::L);
 }
 
+int TestObject::handleTCPMessage(const char *buffer, size_t bufLen) {
```
[the remark this answers is not in the document]
Comment (Angeleon) [reply src/iso22133object.cpp]: remvoed

**Pull request merged**